### ISSUE TITLE
ETH+ZIL mining on rustpool connection issues

### ISSUE THREAD

**Issue by igoralebar** [develsoftware/GMinerRelease#206]
When starting mining initially, everything is fine. 
But then, when it's a ZIL PoW time, miner terminates and restarts after generating DAG for ZIL.
After restart, it generates DAG again and I get "Connection Error: Write time out" and "Stopped Mining on GPU0", after which it reconnects to pool and starts mining ZIL with previously generated DAG.
But after about 30 seconds of mining ZIL inside one minute mining window, I get another "Connection Error: Write time out" and "Stopped Mining on GPU0", after which it reconnects successfully and continues mining with no problems.

Here's part of log with this issue:

```
12:00:47 New Job: 54ddaeea Epoch: #407 Block: #12230798 Diff: 4.00G
12:00:48 New Job: 89e36cda Epoch: #0 Diff: 8.00G
12:00:48 GPU1: Core clock offset changed to default
12:00:48 GPU1: Memory clock offset changed to default
12:00:48 GPU1: Generating DAG for epoch #0 [Single Buffer 1024 MB]
12:00:49 GPU0: Core clock offset changed to default
12:00:49 GPU0: Memory clock offset changed to default
12:00:49 GPU0: Generating DAG for epoch #0 [Single Buffer 1024 MB]
12:00:52 GPU1: DAG generated in 3.19s [321 MB/s]
12:00:52 GPU1: DAG verification completed without errors
12:01:05 Miner terminated, watchdog will restart process after 10 seconds
+----------------------------------------------------------------+
|                          GMiner v2.51                          |
+----------------------------------------------------------------+
Algorithm:          Ethash
DevFee:             0.65%
Stratum server:     
  host:             eu3-zil.rustpool.xyz:8008
  user:            eth_wallet.worker0
  password:         zil_wallet@eth-se.flexpool.io:4444@4G
Power calculator:   on
Color output:       on
Watchdog:           on
API:                *****
Log to file:        log_eth_zil.txt
Selected devices:   GPU0 GPU1
Intensity:          100  100 
Temperature limits: 90C  90C 
------------------------------------------------------------------
12:01:16 GPU0 NVIDIA NVIDIA GeForce RTX 3060 12GB: Kernel #6 Selected
12:01:16 GPU1 NVIDIA NVIDIA GeForce GTX 1080 8GB: Kernel #5 Selected
12:01:16 GPU1: Memory tweak #5 initialized
12:01:16 GPU0: Core clock offset changed to default
12:01:16 GPU0: Memory clock offset changed to default
12:01:16 GPU1: Core clock offset changed to default
12:01:16 GPU1: Memory clock offset changed to default
12:01:16 GPU0: Target Fan temperature changed to 56 C
12:01:16 GPU1: Target Fan temperature changed to 51 C
12:01:16 Connected to eu3-zil.rustpool.xyz:8008 [13.49.168.130]
12:01:16 New Job: b9584216 Epoch: #0 Diff: 8.00G
12:01:16 Started Mining on GPU0: NVIDIA NVIDIA GeForce RTX 3060 12GB [0000:0d:00.0]
12:01:16 Started Mining on GPU1: NVIDIA NVIDIA GeForce GTX 1080 8GB [0000:0e:00.0]
12:01:17 GPU0: Generating DAG for epoch #0 [Single Buffer 1024 MB]
12:01:17 GPU1: Generating DAG for epoch #0 [Single Buffer 1024 MB]
12:01:19 GPU0: DAG generated in 2.26s [452 MB/s]
12:01:19 GPU0: DAG verification completed without errors
12:01:19 GPU0: Core clock offset changed to -280 MHz
12:01:19 GPU0: Memory clock offset changed to +960 MHz
12:01:19 GPU1: DAG generated in 2.53s [405 MB/s]
12:01:19 GPU1: DAG verification completed without errors
12:01:22 GPU1: Core clock offset changed to +163 MHz
12:01:22 GPU1: Memory clock offset changed to +160 MHz
12:01:33 New Job: 3a267389 Epoch: #0 Diff: 8.00G
12:01:46 Connection Error: Write timed out
12:01:46 Stopped Mining on GPU0
12:01:46 Stopped Mining on GPU1
12:01:52 Connected to eu3-zil.rustpool.xyz:8008 [13.49.168.130]
12:01:52 New Job: 04d0035c Epoch: #0 Diff: 8.00G
12:01:52 Started Mining on GPU0: NVIDIA NVIDIA GeForce RTX 3060 12GB [0000:0d:00.0]
12:01:52 Started Mining on GPU1: NVIDIA NVIDIA GeForce GTX 1080 8GB [0000:0e:00.0]
12:02:02 New Job: 4b993309 Epoch: #0 Diff: 8.00G
12:02:03 New Job: 43a41762 Epoch: #0 Diff: 8.00G
12:02:04 New Job: 2523c7dc Epoch: #0 Diff: 8.00G
12:02:05 New Job: 4b993309 Epoch: #0 Diff: 8.00G
12:02:06 New Job: 4e22bdec Epoch: #0 Diff: 8.00G
12:02:09 New Job: 4b993309 Epoch: #0 Diff: 8.00G
12:02:19 New Job: 43a41762 Epoch: #0 Diff: 8.00G
12:02:21 New Job: cb0faf41 Epoch: #0 Diff: 8.00G
12:02:22 Connection Error: Write timed out
12:02:22 Stopped Mining on GPU0
12:02:22 Stopped Mining on GPU1
12:02:27 Connected to eu3-zil.rustpool.xyz:8008 [13.49.168.130]
12:02:27 New Job: ecdf51f2 Epoch: #0 Diff: 8.00G
12:02:27 Started Mining on GPU0: NVIDIA NVIDIA GeForce RTX 3060 12GB [0000:0d:00.0]
12:02:27 Started Mining on GPU1: NVIDIA NVIDIA GeForce GTX 1080 8GB [0000:0e:00.0]
12:02:28 New Job: cb0faf41 Epoch: #0 Diff: 8.00G
12:02:30 New Job: 43a41762 Epoch: #0 Diff: 8.00G
12:02:32 New Job: 4b993309 Epoch: #0 Diff: 8.00G
12:02:34 New Job: e37779b4 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:37 GPU0: Core clock offset changed to default
12:02:37 GPU0: Memory clock offset changed to default
12:02:37 GPU0: Generating DAG for epoch #407 [Single Buffer 4280 MB]
12:02:37 GPU1: Core clock offset changed to default
12:02:37 GPU1: Memory clock offset changed to default
12:02:37 GPU1: Generating DAG for epoch #407 [Single Buffer 4280 MB]
12:02:37 New Job: 5d41d8a9 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:40 New Job: 28461444 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:43 New Job: c8128b29 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:46 New Job: bcaf14b6 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:47 GPU0: DAG generated in 10.29s [416 MB/s]
12:02:47 GPU0: DAG verification completed without errors
12:02:49 GPU1: DAG generated in 11.82s [362 MB/s]
12:02:49 GPU1: DAG verification completed without errors
12:02:49 GPU0: Core clock offset changed to -280 MHz
12:02:49 GPU0: Memory clock offset changed to +960 MHz
12:02:49 GPU1: Core clock offset changed to +163 MHz
12:02:49 GPU1: Memory clock offset changed to +160 MHz
12:02:49 New Job: 90bc03d1 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:52 New Job: a026bee4 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:55 New Job: ba3ffbd7 Epoch: #407 Block: #12230808 Diff: 4.00G
12:02:57 Connection Error: Write timed out
12:02:57 Stopped Mining on GPU0
12:02:57 Stopped Mining on GPU1
12:03:02 Connected to eu3-zil.rustpool.xyz:8008 [13.49.168.130]
12:03:02 Authorized on Stratum Server
12:03:02 New Job: 4e8f8d2e Epoch: #407 Block: #12230808 Diff: 4.00G
12:03:02 Started Mining on GPU0: NVIDIA NVIDIA GeForce RTX 3060 12GB [0000:0d:00.0]
12:03:02 Started Mining on GPU1: NVIDIA NVIDIA GeForce GTX 1080 8GB [0000:0e:00.0]
12:03:02 New Job: 90858eee Epoch: #407 Block: #12230808 Diff: 4.00G
12:03:04 New Job: bb627038 Epoch: #407 Block: #12230808 Diff: 4.00G
12:03:07 New Job: 302879a6 Epoch: #407 Block: #12230808 Diff: 4.00G
12:03:08 GPU0: Share #1 verified on CPU, difficulty: 34.58G
12:03:08 GPU0: Share #1 accepted 34 ms
12:03:10 New Job: 1d10ce59 Epoch: #407 Block: #12230808 Diff: 4.00G
```

**Comment by igoralebar** [comment]
I have to mention that this only happens on the first cycle of ETH+ZIL mining.
Once DAGs for both coins are successfully cached, the switch happens seamlessly.
But it means that I can't restart miner after tweaking some setting without stumbling upon this issue again, which steals precious seconds from being able to get ZIL shares.

**Comment by igoralebar** [comment]
found that Hardware Accelerated GPU Scheduling is what's causing this issue. Once I disable HAGS, nothing like that happens.

**Comment by igoralebar** [comment]
still happens even without HAGS when DAG is generated for new epoch

**Comment by igoralebar** [comment]
Fixed in version 2.53
